src/Microsoft.Health.Operations.Functions.UnitTests/Management/PurgeOrchestrationInstanceHistoryTests.cs

```diff
@@ -43,14 +45,57 @@ public PurgeOrchestrationInstanceHistoryTests()
     [Theory]
     [InlineData(0)]
     [InlineData(12)]
-    public async Task GivenNoOrchestrationInstances_WhenPurgeCompletedDurableFunctionsHistory_ThenNoOrchestrationsPurgedAsync(int deleted)
+    public async Task GivenOrchestrationInstances_WhenPurgeCompletedDurableFunctionsHistory_ThenOrchestrationsPurgedAsync(int count)
     {
+        var instanceId = Guid.NewGuid().ToString();
+
+        var durableOrchestrationState = Enumerable.Repeat(new DurableOrchestrationStatus { InstanceId = instanceId }, count);
+
         _durableClient
-            .PurgeInstanceHistoryAsync(
-                DateTime.MinValue,
-                _utcNow.AddDays(-_purgeConfig.MinimumAgeDays),
-                _purgeConfig.Statuses)
-            .Returns(new PurgeHistoryResult(deleted));
+            .ListInstancesAsync(Arg.Is<OrchestrationStatusQueryCondition>(condition => AreConditionEqual(condition)), Arg.Any<CancellationToken>())
+            .Returns(new OrchestrationStatusQueryResult
+            {
+                DurableOrchestrationState = durableOrchestrationState
+            });
+
+        _durableClient
+            .PurgeInstanceHistoryAsync(instanceId)
+            .Returns(new PurgeHistoryResult(count));
+
+        using (Mock.Property(() => ClockResolver.UtcNowFunc, () => _utcNow))
+        {
+            await _purgeTask.Run(_timer, _durableClient, NullLogger.Instance);
+        }
+
+        await _durableClient
+            .Received(count)
+            .PurgeInstanceHistoryAsync(instanceId);
+    }
+
+    [Fact]
+    public async Task GivenOrchestrationInstancesAndInstancesToSkipPurging_WhenPurgeCompletedDurableFunctionsHistory_ThenNoOrchestrationsPurgedAsync()
+    {
+        var instanceId1 = Guid.NewGuid().ToString();
+        var instanceId2 = Guid.NewGuid().ToString();
+
+        _purgeConfig.InstancesToSkipPurging = new string[] { instanceId1 };
+
+        var durableOrchestrationState = new List<DurableOrchestrationStatus> 
+        {
+            new DurableOrchestrationStatus { InstanceId = instanceId1, Name = instanceId1 },
```

Comment:
This test is a bit confusing because you're using the instance Id as the name. Can you change it to reflect an actual name? Otherwise it looks like the options are used to filter out instance IDs

Reply:
done